### ISSUE TITLE
Error building dev env on Windows

### ISSUE THREAD

**Issue by Paulo-Weber** [Haufe-Lexware/wicked.haufe.io#262]
I have a similar error as the described on [this issue](https://github.com/Haufe-Lexware/wicked.haufe.io/issues/252).
The portal can't find the access token, 
![image](https://user-images.githubusercontent.com/14895907/74840893-871dae80-5306-11ea-9293-b30f9956ca9c.png)

I had tried the `build.sh` script, as the `npm install` on all the subfolders (and `./install.sh`) but it still didn't work.

Logs from ui-error.log:
```
Error: connect ECONNREFUSED 10.1.27.8:3001
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1054:14)
checkConfigHash(): An error occurred.
Error: connect ECONNREFUSED 10.1.27.8:3001
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1054:14) {
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '10.1.27.8',
  port: 3001
}
Error: connect ECONNREFUSED 10.1.27.8:3001
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1054:14)
Environment variable PORTAL_API_URL is not set, defaulting to http://10.1.27.8:3001/. If this is not correct, please set before starting this process.
Error: Did not receive anonymous access token.
    at Request._callback (C:\Repos\wicked.haufe.io\src\ui\routes\utils.js:337:29)
    at Request.self.callback (C:\Repos\wicked.haufe.io\src\ui\node_modules\request\request.js:185:22)
    at Request.emit (events.js:196:13)
    at Request.<anonymous> (C:\Repos\wicked.haufe.io\src\ui\node_modules\request\request.js:1161:10)
    at Request.emit (events.js:196:13)
    at IncomingMessage.<anonymous> (C:\Repos\wicked.haufe.io\src\ui\node_modules\request\request.js:1083:12)
    at Object.onceWrapper (events.js:284:20)
    at IncomingMessage.emit (events.js:201:15)
    at endReadableNT (_stream_readable.js:1130:12)
    at processTicksAndRejections (internal/process/task_queues.js:83:17)
Error: Not Found
    at C:\Repos\wicked.haufe.io\src\ui\app.js:189:21
    at Layer.handle [as handle_request] (C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\index.js:317:13)
    at C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\index.js:284:7
    at Function.process_params (C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\index.js:335:12)
    at next (C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\index.js:275:10)
    at C:\Repos\wicked.haufe.io\src\ui\app.js:165:16
    at Layer.handle [as handle_request] (C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\layer.js:95:5)
    at trim_prefix (C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\index.js:317:13)
    at C:\Repos\wicked.haufe.io\src\ui\node_modules\express\lib\router\index.js:284:7 {
  status: 404
}

```

**Comment by DonMartin76** [comment]
Which environment are you using? Windows 10? WSL? Can you please be as specific as possible, then I will try to reproduce.

Admittedly, most of us develop on macOS, but there is me knowingly no real blockers which should prevent you from using Windows. It will be some kind of regression.

Sorry for the inconvenience, will try to reproduce as soon as I know which env you are using.

**Comment by DonMartin76** [comment]
Extra question: Are you on the `next` branch of the repo?

**Comment by Paulo-Weber** [comment]
I'm running on Windows 10 Pro 64bits, not a WSL. And using the branch `next`.

Prior to this error, I was not being able to run the services that use `npm run ...`, because the windows was trying to somehow open the npm.cmd, instead of executing it. So I changed the `wicked-pm2.config.js`  file, editing all scripts that run npm, to run a bash script that will run `npm run ...` So after that I got this error, I'm not sure if this one is related to another.

**Comment by DonMartin76** [comment]
OK. I will need to look into this. The dev env is currently certainly **not** optimized for development on Windows. But not saying this must stay like that, we just haven't tried it out.

Running the wicked-cli and the wicked box works though?

**Comment by DonMartin76** [comment]
Sorry for not responding - I actually had - but GitHub was stuck in some weird limbo state and never accepted my comment.

**Comment by Paulo-Weber** [comment]
wicked-cli and wicked box worked just fine. So it's something only on the development environment.

**Comment by Paulo-Weber** [comment]
@DonMartin76 Any update on this Issue?

**Comment by DonMartin76** [comment]
Not yet, no, sorry. I was busy with doing the current release. I'll try to find some time in the next days, but my Windows machine is hopelessly outdated :-D

**Comment by DonMartin76** [comment]
Amid this Corona thing I have finally managed to arrange for myself a Windows computer in my home office and can now look at this. Grmbl.

**Comment by DonMartin76** [comment]
I have spent quite some time with this, and I am starting to think it's probably not going to work to run the development environment directly on Windows. Which bash shell have you been using? I have been trying with  git bash (mingw) and got pretty far. But then I run into really bad issues.

I will give a WSL setup a try, using the docker engine from Windows via the docker sock, and use a native Ubuntu node installation instead. Let's see whether that works.

**Comment by Paulo-Weber** [comment]
I used the mingw as well, and I haven't tried a WSL setup.

**Comment by DonMartin76** [comment]
I can now reproduce the issue, and I have some ideas what the reason is. It seems to - under certain circumstances - resolve the wrong local IP address. I get two of them, and the scripts pick the wrong one. Using WSL for now, and that works nice in the rest of the aspects. I will try to fix the IP address thing, and then update the docs to work via WSL on Windows.

**Comment by DonMartin76** [comment]
# Set up dev env on Windows

(I will add this to the official documentation; if you still are having issues, please respond here oor file a new issue)

* Install Docker for Windows
    * https://docs.docker.com/docker-for-windows/install/
* Install WSL
    * e.g., https://www.windowscentral.com/install-windows-subsystem-linux-windows-10
    * Running wsl.exe must open a bash terminal window
* Install docker command line tools inside WSL
    * For example: https://nickjanetakis.com/blog/setting-up-docker-for-windows-and-wsl-to-work-flawlessly
    * Includes: Open up Docker for Windows via tcp://localhost:2375 
    * Link DOCKER_HOST on WSL to above docker instance
    * There is also a way to do this securely, but this works, IF you have a safe network environment - don’t do this on a public network
    * In the end, the docker command must work just like on Windows (or on any other machine), from inside the WSL
    * Known caveat: WSL uses .bashrc instead of the suggested .bash_profile; add the export command to .bashrc instead of creating a new .bash_profile (which will kill everything else)
* Install nvm
    * https://github.com/nvm-sh/nvm
* Install node.js 12: nvm install 12 && nvm use 12
* Install pm2 globally; this is used for running the dev env: npm install -g pm2
* If you have not already: Install VS Code
    * Install VS Code WSL Remote tools, to be able to work on the files inside the WSL subsystem
* Clone wicked repo *inside* WSL home directory
    * git clone https://github.com/Haufe-Lexware/wicked.haufe.io
    * git checkout next
* Then set up dev env
    * cd src/tools/develop
    * ./build-kong.sh
    * ./install.sh
    * ./start-devenv.sh
* Your local development version of wicked should now run on localhost:3000

**NOTE**: This is not yet adapted to WSL2; this still uses the "old" firs version of WSL.

**Comment by DonMartin76** [comment]
In the docs now: https://github.com/Haufe-Lexware/wicked.haufe.io/blob/next/doc/development-environment.md

Closing this. Feel free to reopen or create a new issue in case you find other things.
